compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -2617,6 +2623,7 @@ impl<'tcx> SelectionContext<'_, 'tcx> {
     ) -> Result<Vec<PredicateObligation<'tcx>>, ()> {
         self.infcx
             .at(&obligation.cause, obligation.param_env)
+            // ICEs if we use `DefineOpaqueTypes::Yes` here, see #116652.
```

Comment:
What *kind* of ICE? Where? There's no need to link #116652, but there does need to be more information here.